src/debugging-support-in-rustc.md

```diff
@@ -268,6 +268,36 @@ Focus is to let macros decide what to do. This can be achieved by having some ki
 that lets the macro tell the compiler where the line marker should be. This affects where you
 set the breakpoints and what happens when you step it.
 
+## Source file checksums in debug info
+
+Both DWARF and CodeView (PDB) support embedding a cryptographic hash of each source file that
+contributed the associated binary.
+
+The cryptographic hash can be used by a debugger to verify that the source file matches the
+executable. If the source file does not match, the debugger can provide a warning to the user.
+
+The hash can also be used to prove that a given source file has not been modified since it was
+used to compile an executable. Because MD5 and SHA1 both have demonstrated vulnerabilities,
+using SHA256 is recommended for this application.
```

Comment:
It might be worth noting roughly where in the code this information is attached -- e.g., the name of a function or module. Is this something that is done in the LLVM-specific code, or is it portable across backends by default?

Reply:
I added some additional detail. It is portable across backends.